ledger-service/http-json/src/main/scala/com/digitalasset/http/EndpointsCompanion.scala

```diff
@@ -41,27 +40,32 @@ object EndpointsCompanion {
 
   final case class Unauthorized(message: String) extends Error
 
-  final case class ServerError(message: String) extends Error
+  final case class ServerError(message: Throwable \/ String) extends Error
 
   final case class ParticipantServerError(grpcStatus: GrpcCode, description: Option[String])
       extends Error
 
   final case class NotFound(message: String) extends Error
 
+  object ServerError {
+    def apply(message: String): ServerError = ServerError(\/.right(message))
+    def apply(error: Throwable): ServerError = ServerError(\/.left(error))
```

Comment:
Here also just do `error.message`

Reply:
I might be missing something here, but a ServerError with a message or an error, they are both indicating an error, right? What is the distinction for?